### PR TITLE
Add Jest for testing

### PR DESCRIPTION
**Pull request by t-kelly**
### What are you trying to accomplish with this PR?
Fixes https://github.com/Shopify/slate/issues/180

For maintainers:
- [ ] I have :tophat:'d these changes.
- [ ] I have bumped the `package.json` version in a separate PR, if applicable.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  globals: {
+    slate: {}
+  },
+};
```
Comment (m-ux) [inline jest.config.js]: newline

**Comment by t-kelly** [comment]
Looks like CircleCI will call `npm test` automatically [when test is included as a script in `package.json`](https://circleci.com/docs/1.0/language-nodejs/#testing)

**Comment by t-kelly** [comment]
Actually I'll need to change `slate test` from [`override` to `post`](https://circleci.com/docs/1.0/configuration/#phases)

**Comment by m-ux** [comment]
just wanted to point out that, even though this is a super basic file, these are good tests :+1:

Way too often in Shopify we test the implementation and not the interface.  Stick to testing the results of interactions with the interface methods.  Focus on inputs & outputs - don't bother exposing or testing private methods directly - test them inferentially through expectations on the interface.  

You guys are on the right path here, keep it up 👏

**Pull request merged**